ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -0,0 +1,348 @@
+        Arrays.asList(
+            "incremental",
+            String.valueOf(work.eventFrom),
+            String.valueOf(lastReplId)
+        ),
+        dmd.getDumpFilePath(), conf);
+    dmd.setDump(DumpType.INCREMENTAL, work.eventFrom, lastReplId, cmRoot);
+    dmd.write();
+    return lastReplId;
+  }
+
+  private void dumpEvent(NotificationEvent ev, Path evRoot, Path cmRoot) throws Exception {
+    EventHandler.Context context = new EventHandler.Context(
+        evRoot,
+        cmRoot,
+        getHive(),
+        conf,
+        getNewEventOnlyReplicationSpec(ev.getEventId())
+    );
+    EventHandlerFactory.handlerFor(ev).handle(context);
+    REPL_STATE_LOG.info(
+        "Repl Dump: Dumped event with ID: {}, Type: {} and dumped metadata and data to path {}",
+        String.valueOf(ev.getEventId()), ev.getEventType(), evRoot.toUri().toString());
+  }
+
+  private ReplicationSpec getNewEventOnlyReplicationSpec(Long eventId) throws SemanticException {
+    return getNewReplicationSpec(eventId.toString(), eventId.toString());
+  }
+
+  private Long bootStrapDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot) throws Exception {
+    // bootstrap case
+    Long bootDumpBeginReplId = getHive().getMSC().getCurrentNotificationEventId().getEventId();
+    for (String dbName : matchesDb()) {
+      REPL_STATE_LOG
+          .info("Repl Dump: Started analyzing Repl Dump for DB: {}, Dump Type: BOOTSTRAP",
+              dbName);
+      LOG.debug("ReplicationSemanticAnalyzer: analyzeReplDump dumping db: " + dbName);
+
+      Path dbRoot = dumpDbMetadata(dbName, dumpRoot);
+      dumpFunctionMetadata(dbName, dumpRoot);
+      for (String tblName : matchesTbl(dbName, work.tableNameOrPattern)) {
+        LOG.debug(
+            "analyzeReplDump dumping table: " + tblName + " to db root " + dbRoot.toUri());
+        dumpTable(dbName, tblName, dbRoot);
+      }
+    }
+    Long bootDumpEndReplId = getHive().getMSC().getCurrentNotificationEventId().getEventId();
+    LOG.info("Bootstrap object dump phase took from {} to {}", bootDumpBeginReplId,
+        bootDumpEndReplId);
+
+    // Now that bootstrap has dumped all objects related, we have to account for the changes
+    // that occurred while bootstrap was happening - i.e. we have to look through all events
+    // during the bootstrap period and consolidate them with our dump.
+
+    IMetaStoreClient.NotificationFilter evFilter =
+        new DatabaseAndTableFilter(work.dbNameOrPattern, work.tableNameOrPattern);
+    EventUtils.MSClientNotificationFetcher evFetcher =
+        new EventUtils.MSClientNotificationFetcher(getHive().getMSC());
+    EventUtils.NotificationEventIterator evIter = new EventUtils.NotificationEventIterator(
+        evFetcher, bootDumpBeginReplId,
+        Ints.checkedCast(bootDumpEndReplId - bootDumpBeginReplId) + 1,
+        evFilter);
+
+    // Now we consolidate all the events that happenned during the objdump into the objdump
+    while (evIter.hasNext()) {
```

Comment:
yeh i was not sure when it was required so i have left it for now to be a gentle reminder, may be we discuss this in next catch up and create a separate task for this.